ui/src/main/webapp/home/reducer.js

```diff
@@ -1,10 +1,12 @@
 import { combineReducers } from 'redux-immutable'
 import { Map } from 'immutable'
+import sub from 'redux-submarine'
 
-export const getSourceConfigs = (state) => state.getIn(['home', 'configs', 'sources'])
-export const getLdapConfigs = (state) => state.getIn(['home', 'configs', 'ldap'])
-export const getSubmittingPids = (state) => state.getIn(['home', 'submitting']).toJS()
-export const getConfigErrors = (state, servicePid) => state.getIn(['home', 'errors', servicePid], [])
+export const submarine = sub()
+export const getSourceConfigs = (state) => submarine(state).getIn(['configs', 'sources'])
```

Comment:
The root reducer now passes all of that information down to these sub-reducers via redux-submarine so that they can work without having to know where the root reducer put them in the state tree. If you look at the root reducer, they import this file's submarine and `init` it to 'home' (which is where it decided to nest this sub-reducer) so when this file calls `submarine(state)`, it will return only the 'home' part of the state tree so that this sub-reducer's selectors can operate as if it was the only thing in the state tree.